### PR TITLE
Utility to convert a string (e.g. 100MB) to a number of bytes

### PR DESCRIPTION
**Pull request by Tempate**
Convert to bytes a string with a floating number followed by either the letters B, KB, MB, GB, TB, PB or KiB, MiB, GiB, TiB, PiB (case insensitive).

### CODE REVIEW

File: cpp_utils/src/cpp/utils.cpp [hunk 1]
```diff
@@ -99,6 +100,47 @@ void to_uppercase(
             });
 }
 
+unsigned long long to_bytes(const std::string& input)
```
Comment (EduPonz) [inline cpp_utils/src/cpp/utils.cpp:103]: I'd suggest using explicit size types, in this case `uint64_t`

File: cpp_utils/src/cpp/utils.cpp [hunk 2]
```diff
@@ -99,6 +100,47 @@ void to_uppercase(
...
+    double number = std::stod(number_str);
+
+    // Extract the magnitude
+    std::string magnitude_str = matches[2].str();
+    to_uppercase(magnitude_str);
+
+    const auto magnitude = magnitudes.at(magnitude_str);
+
+    // Calculate the number of bytes
+    const unsigned long long bytes = number * magnitude;
```
Comment (EduPonz) [inline cpp_utils/src/cpp/utils.cpp:139]: This could overflow, I'd add a check to prevent it. For example what if the input it `9999999999PiB`

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: cpp_utils/src/cpp/utils.cpp [hunk 3]
```diff
@@ -99,6 +100,55 @@ void to_uppercase(
             });
 }
 
+std::uint64_t to_bytes(
```
Comment (EduPonz) [inline cpp_utils/src/cpp/utils.cpp:103]: Include cstdint

File: cpp_utils/src/cpp/utils.cpp [hunk 4]
```diff
@@ -99,6 +100,55 @@ void to_uppercase(
...
+    const auto magnitude = magnitudes.at(magnitude_str);
+
+    // Check for overflow
+    if (number > std::numeric_limits<std::uint64_t>::max() / magnitude)
+    {
+        throw std::invalid_argument("The number is too large to be converted to bytes.");
+    }
+
+    // Calculate the number of bytes
+    const std::uint64_t bytes = number * magnitude;
```
Comment (EduPonz) [inline cpp_utils/src/cpp/utils.cpp:147]: There is a loss of precision here as number being a double, the cast to `uint64_t` will be an approximate of the real input, although I don't know if there is something we can do about it, given that there is no `std::stoui64` 🥲. In that case, I think this should be:
```suggestion
    // Explicit cast to uint64_t is safe here as the number has already been checked to fit.
    // Furthermore, the product is also safe as the possible overflown has also been checked before
    const std::uint64_t bytes = static_cast<std::uint64_t>(number) * magnitude;
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Review by EduPonz — DISMISSED** [review]

**Review by LuciaEchevarria99 — CHANGES_REQUESTED** [review]

**Review by LuciaEchevarria99 — DISMISSED** [review]

**Review by LuciaEchevarria99 — APPROVED** [review]
LGTM

**Pull request merged**